bundles/org.openhab.binding.resol/src/main/java/org/openhab/binding/resol/handler/ResolBridgeHandler.java

```diff
@@ -0,0 +1,522 @@
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.resol.handler;
+
+import java.io.IOException;
+import java.net.InetAddress;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.jdt.annotation.NonNull;
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.eclipse.jdt.annotation.Nullable;
+import org.openhab.binding.resol.internal.ResolBindingConstants;
+import org.openhab.binding.resol.internal.ResolBridgeConfiguration;
+import org.openhab.binding.resol.internal.discovery.ResolDiscoveryService;
+import org.openhab.binding.resol.internal.providers.ResolChannelTypeProvider;
+import org.openhab.core.i18n.LocaleProvider;
+import org.openhab.core.thing.Bridge;
+import org.openhab.core.thing.Channel;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.Thing;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.thing.binding.BaseBridgeHandler;
+import org.openhab.core.thing.binding.builder.ChannelBuilder;
+import org.openhab.core.thing.binding.builder.ThingBuilder;
+import org.openhab.core.thing.type.ChannelTypeUID;
+import org.openhab.core.types.Command;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import de.resol.vbus.Connection;
+import de.resol.vbus.Connection.ConnectionState;
+import de.resol.vbus.ConnectionAdapter;
+import de.resol.vbus.Packet;
+import de.resol.vbus.Specification;
+import de.resol.vbus.Specification.PacketFieldValue;
+import de.resol.vbus.SpecificationFile;
+import de.resol.vbus.SpecificationFile.Language;
+import de.resol.vbus.TcpDataSource;
+import de.resol.vbus.TcpDataSourceProvider;
+
+/**
+ * The {@link ResolBridgeHandler} class handles the connection to the VBUS/LAN adapter.
+ *
+ * @author Raphael Mack - Initial contribution
+ */
+@NonNullByDefault
+public class ResolBridgeHandler extends BaseBridgeHandler {
+
+    private final Logger logger = LoggerFactory.getLogger(ResolBridgeHandler.class);
+
+    private Language lang;
+    private Locale locale;
+
+    private String ipAddress = "";
+    private String password = "";
+    private int refreshInterval;
+    private boolean isConnected = false;
+    private String unconnectedReason = "";
+
+    // Background Runables
+
+    @Nullable
+    private ScheduledFuture<?> pollingJob;
+
+    @Nullable
+    private Connection tcpConnection;
+    private Specification spec;
+    private Set<String> availableDevices = new HashSet<String>();
+
+    private Map<String, ResolThingHandler> thingHandlerMap = new HashMap<String, ResolThingHandler>();
```

Comment:
You probably should not keep a map of things. The bridge already tracks this itself via children. So you're re-implementing part of what is already available, making this code more complex than needed. See if you can change the code to make use of this functionality.

Reply:
I feel stupid or blind right now :grinning:, can you please point me to it?

Reply:
You can get it via `getThing().getThings()` see for example https://github.com/openhab/openhab-addons/blob/b6eea715fa74e630283726d603f5e24f9f315a6f/bundles/org.openhab.binding.spotify/src/main/java/org/openhab/binding/spotify/internal/handler/SpotifyBridgeHandler.java#L427